src/reaction_network.jl

```diff
@@ -281,7 +281,7 @@ function make_func(func_expr::Vector{Expr},reactants::OrderedDict{Symbol,Int},pa
     system = Expr(:block)
     for func_line in deepcopy(func_expr)
         tmp_line = recursive_replace!(func_line, (reactants,:internal_var___u), (parameters, :internal_var___p))
-        push!(system.args, :(@inbounds $tmp_line))
+        push!(system.args, :(inbounds_func($tmp_line)))
```

Comment:
this doesn't seem to make sense. If you already are doing the `tmp` line, then you already indexed and thus already needed to `@inbounds`?

Reply:
Correct me if I am wrong (well somewhere I am obviously wrong of you ask Travis). Here `tmp_line` is an expression which have not actually been executed yet. `tmp_line` might be something like
```julia
internal_var___du[1] = internal_var___p[1] - internal_var___p[2]*internal_var___u[1]
```
Next the following expression
```julia
inbounds_func(internal_var___du[1] = internal_var___p[1] - internal_var___p[2]*internal_var___u[1])
```
is pushed on an expression array. But that actual line is not execute until the last line of the entire `@reaction_network` macro.

I have not been able to determine what the error is here on git, but I can run stuff on my machine and get a `UndefVarError: inbounds_func not defined` error when using the reaction network as input to a solver. This suggests that the program is trying to execute the ` inbounds_fun` after the model has been made.

Could be solved by exporting the ` inbounds_fun` function. However, it I have understood anything then the `@inbounds` stuff is supposed to be a direction to the compiler? Then should it actually have to be called in the solver?

Reply:
This would execute the argument expression `tmp_line` and then execute the function, but when `tmp_line` is executed that's when it will need `@inbounds`. It won't put that expression into the function. But I found out last night that the `@inbounds` not returning bug was fixed on v0.7, so we might want to just pull back a commit and then postpone for v0.7.

Reply:
Understood.

I agree, if the bug is fixed in a later version then we wait for that. Is there a simple way to remove the last commit from the chain. Or should I simply redo and add another commit?

Reply:
You can give a revert commit or branch back and force push. A revert commit is probably easier (and safer)